spec/features/editing_the_trademark_decision_filters_and_options_spec.rb

```diff
@@ -0,0 +1,71 @@
+require "spec_helper"
+require "gds_api/test_helpers/support_api"
+
+RSpec.feature "Editing the Trademark Decisions finder filters and options", type: :feature do
+  include GdsApi::TestHelpers::SupportApi
+
+  let(:organisations) do
+    [
+      { "content_id" => "5d6f9583-991f-413d-ae83-be7274e5eae4", "title" => "Intellectual Property Office (IPO)" },
+    ]
+  end
+
+  before do
+    Capybara.current_driver = Capybara.javascript_driver
+    log_in_as_editor(:gds_editor)
+    stub_publishing_api_has_content([], hash_including(document_type: TrademarkDecision.document_type))
+    stub_publishing_api_has_content(organisations, hash_including(document_type: Organisation.document_type))
+    stub_any_support_api_call
+  end
+
+  scenario "editing, deleting and adding new filters" do
+    visit "admin/trademark-decisions"
+    within "#facets_summary_card" do
+      click_link "Request changes"
+    end
+
+    expect(page).to have_selector("span", text: "Trademark Decision finder")
+    expect(page).to have_selector("h1", text: "Request change: Filters and options")
+
+    # Delete all but the first filter
+    all(".js-add-another__remove-button").last.click while all(".js-add-another__remove-button").count > 1
+    click_button "Add another filter"
+    within find("fieldset", text: "Filter 2") do
+      # 2 in "Filter 2" is the visible index of the newly inserted filter, since we deleted all but the first one, and now we're adding a new one
```

Comment:
I think the code is pretty much self-documenting, but we can move your comment above and shorten it to something like:

```suggestion
    # Fill in the newly created filter
    within find("fieldset", text: "Filter 2") do
```

Reply:
It was the indices that were tripping me. Better off with no comment.